src/gflownet/data/sampling_iterator.py

```diff
@@ -81,18 +89,21 @@ def __len__(self):
 
     def __iter__(self):
         worker_info = torch.utils.data.get_worker_info()
-        wid = (worker_info.id if worker_info is not None else 0)
-        self.rng = self.algo.rng = self.task.rng = np.random.default_rng(142857 + wid)
+        self._wid = (worker_info.id if worker_info is not None else 0)
+        self.rng = self.algo.rng = self.task.rng = np.random.default_rng(142857 + self._wid)
         self.ctx.device = self.device
+        if self.log_dir is not None:
+            os.makedirs(self.log_dir, exist_ok=True)
+            self.generated_logfile = open(f'{self.log_dir}/generated_mols_{self._wid}.csv', 'a')
```

Comment:
The intent here is to open a file and gradually append to it as new objects are being generated by the model. What would be the right way to handle this? Reopening the log file in `'a'` mode after every minibatch? 

Note that I'm thinking of switching to `sqlite` instead, since it will scale much better and more gracefully than a csv, and presumably handle the underlying opening and closing of files better.

Reply:
Moving to sqlite might be a good idea. As for logging to the file, I'd do a `with open` in `a` mode directly when you are ready to write in the `log_generated` method. This ensure the file handle does get left open which is never a good idea. The other option would be to call `close()` explicitely at some point toward the end of the run but this is not a great idea: your program might fail before that, etc.

Reply:
Now addressed by #23.